docs/design/datacontracts/Thread.md

```diff
@@ -240,4 +248,47 @@ TargetPointer IThread.GetThreadLocalStaticBase(TargetPointer threadPointer, Targ
     }
     return threadLocalStaticBase;
 }
+
+byte[] IThread.GetWatsonBuckets(TargetPointer threadPointer)
+{
+    TargetPointer readFrom = TargetPointer.Null;
+    TargetPointer exceptionTrackerPtr = _target.ReadPointer(threadPointer + /*Thread::ExceptionTracker offset */);
+    if (exceptionTrackerPtr == TargetPointer.Null)
+        return Array.Empty<byte>();
+    TargetPointer thrownObjectHandle = target.ReadPointer(exceptionTrackerPtr + /* ExceptionInfo::ThrownObjectHandle offset */);
+    TargetPointer throwableObjectPtr = target.ReadPointer(thrownObjectHandle);
+    if (throwableObjectPtr != TargetPointer.Null)
+    {
+        TargetPointer watsonBuckets = target.ReadPointer(throwableObjectPtr + /* Exception::WatsonBuckets offset */);
+        if (watsonBuckets != TargetPointer.Null)
+        {
+            uint baseSize = target.ReadPointer(target.ReadPointer(watsonBuckets + /* Object::m_pMethTab offset */) + /* MethodTable::BaseSize offset */);
```

Comment:
We've got the Object.GetArrayData() contract API already defined. It looks like this contract is unnecessarily inlining the same algorithm.